admin_solutions/authentication.adoc

```diff
@@ -5,10 +5,303 @@ toc::[]
+_\https://<master>/oauth/authorize_ via an authenticating proxy. You can configure
+the OAuth server to redirect unauthenticated requests to the proxy.
+
+[[configuring-request-header-auth-on-master]]
+=== Configuring Authentication on the Master
+
+. If you have:
+.. Already completed the installation of Openshift, then copy the
+*_/etc/origin/master/master-config.yaml_* file into a new directory; for example:
++
+----
+$ mkdir reqheadauthconfig; cp master-config.yaml reqheadauthconfig
+----
+.. Not yet installed {product-title}, then start the {product-title} API server,
+specifying the hostname of the (future) {product-title} master and a directory
+to store the configuration file created by the start command:
++
+----
+$ openshift start master --public-master=<apiserver> --write-config=<directory>
+----
++
+For example:
++
+----
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=reqheadauthconfig
+----
++
+[NOTE]
+====
+If you are installing with Ansible, then you must add the
+`*identityProvider*` configuration to the Ansible playbook.
+If you use the following steps to modify your configuration manually after installing with Ansible, then you will lose any modifications whenever you re-run the install tool or upgrade.
+====
++
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. View
+xref:../install_config/configuring_authentication.adoc#reqhead-auth-example-config[the example `*RequestHeaderIdentityProvider*` configuration] 
+and use it as a guide to replace the existing stanza.
+. Modify the `*identityProviders*` stanza based on which headers you plan to
+pass in.
+.. Set the provider `*name*` to something unique and relevant to your
+deployment. This name is prefixed to the returned user ID to form an identity
+name.
+.. If required,
+xref:../install_config/configuring_authentication.adoc#mapping-identities-to-users[change `*mappingMethod*`] 
+to control how mappings are established between the provider's identities and
+user objects.
+.. Set the `*challenge*` parameter to *true* to redirect unauthenticated
+requests from clients expecting `WWW-Authenticate` challenges. 
+.. Set the `*provider.challengeURL*` parameter to the proxy URL to which to send
+clients expecting `WWW-Authenticate` challenges. 
+This parameter can include the 
+xref:../install_config/configuring_authentication.adoc#RequestHeaderIDP-urlquerytokens[`${url}` and `${query}` tokens]
+in the query portion of the URL. 
+.. Set the `*login*` parameter to *true* to redirect unauthenticated requests
+from clients expecting login flows. 
+.. Set the `*provider.loginURL*` parameter to the proxy URL to which to send
+clients expecting login flows. This parameter can include the 
+xref:../install_config/configuring_authentication.adoc#RequestHeaderIDP-urlquerytokens[`${url}` and `${query}` tokens]
+in the query portion of the URL. 
+.. Optionally, change the `*clientCA*` parameter to check incoming requests for
```

Comment:
This should absolutely be done. Without checking the request containing user info in headers against a client CA, anyone that can hit the API server can assume any identity they want